File: api/src/models/membership.py
from datetime import datetime
from pydantic import BaseModel
from enum import Enum

class Role(str, Enum):
    ADMIN = "admin"
    MEMBER = "member"

class Membership(BaseModel):
    membership_id: int
    user_id: int
    room_id: int
    role: Role
    joined_at: datetime
    is_active: bool = True

class MembershipCreateRequest(BaseModel):
    user_id: int
    room_id: int
    role: Role = Role.MEMBER

File: client/models/Membership.ts
export enum Role {
  ADMIN = "admin",
  MEMBER = "member",
}

export interface Membership {
  membershipId: number;
  userId: number;
  roomId: number;
  role: Role;
  joinedAt: string;
  isActive: boolean;
}

export interface MembershipCreateRequest {
  userId: number;
  roomId: number;
  role?: Role;
}

export interface MembershipUpdateRequest {
  membershipId: number;
  role?: Role;
}

export interface RoomMember {
  userId: number;
  name: string;
  membershipId: number;
  role: Role;
  joinedAt: string;
}
